server/src/main/java/io/spine/server/aggregate/AggregateRepository.java

```diff
@@ -426,7 +426,9 @@ public DispatchOutcome dispatchEvent(EventEnvelope event) {
 
     private Set<I> route(EventEnvelope event) {
         var route = route(eventRouting(), event);
-        return route.orElse(ImmutableSet.of());
+        @SuppressWarnings("unchecked")
+        var result = (Set<I>) route.orElse(ImmutableSet.of());
```

Comment:
You can just create an immutable set with a generic parameter. No need to cast here:

```java
return route.orElse(ImmutableSet.<I>of());
```

Reply:
As we investigated, this is an issue with covariance between Kotlin and Java. We're safe to suppress here. It is better to have `Set<I>` here rather than `Set<? extends I>` because we have a fixed identifier type (`I`) per entity class.